app/models/spree/order.rb

```diff
@@ -756,10 +756,14 @@ def ensure_customer
       return if associate_customer
 
       customer_name = bill_address.andand.full_name
-      self.customer = Customer.create(enterprise: distributor, email: email_for_customer,
-                                      user: user, name: customer_name,
-                                      bill_address: bill_address.andand.clone,
-                                      ship_address: ship_address.andand.clone)
+      self.customer = Customer.create!(
```

Comment:
Good point! The create could fail. I'm a bit nervous about this failing the whole checkout instead of just reporting to Bugsnag. But it's probably worth it for the investigation.

Reply:
yes, but if this was the reason, it's not happening a whole lot. At least, that we know of.

Reply:
I'm a bit nervous as well...

The "soft" option here would be to drop the bang! and add:
```ruby
Bugsnag.notify("Customer not created with order") unless customer.persisted?
```
that way if we encounter this we'll get an alert with a backtrace instead of a potentially checkout-blocking error, right?

Reply:
fair enough. It's not that the current situation is much better than a checkout-blocking error. At least, as is, the hub doesn't miss a "conversion", that's true.

Reply:
@Matt-Yorkley I just changed the approach as you mentioned. Let me know what you think. If we like it might squash some commits before moving on.